File: 5.implementation/NXP/redsun-service/src/main/java/com/redsun/service/service/MilestonesService.java
package com.redsun.service.service;

import com.redsun.service.entities.Milestones;
import com.redsun.service.entities.Result;

/**
 * Milestones Service interface
 */
public interface MilestonesService {
	
    // Insert.
	Result insert(final Milestones milestones) ;

    // Update.
	Result update(final Milestones milestones) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listMilestonesForPageAndFilter(final int itemsPerPage, final int pageNo, final Milestones milestones);
	
	// List by projectId.
	Result listMilestonesByProjectId(final int projectId);

}

File: 5.implementation/redsun/src/main/webapp/static/js/services/MilestonesService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('milestoneService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(milestone) {
			var urlServer = $rootScope.makePostURL("/milestone/insert");
			return $http.post(urlServer, milestone);
		}
		
		// Update.
		, update: function(milestoneId, milestone) {
			var urlServer = $rootScope.makePostURL("/milestone/update/" + milestoneId);
			return $http.put(urlServer, milestone);
		}
		
		// Delete.
		, delete: function(milestoneId) {
			var urlServer = $rootScope.makePostURL("/milestone/delete/" + milestoneId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(milestoneId) {
			var url = $rootScope.makeGetURL("/milestone/getbyid/" + milestoneId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, milestone) {
			var url = $rootScope.makeGetURL("/milestone/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: milestone });
		}
		
	}// return.
});